app/templates/components/event-location.hbs

```diff
@@ -0,0 +1,21 @@
+{{#if @data.event.online}}
+ {{#if @data.event.locationName}}
+  <span class="{{if @orderPage 'weight-700'}}">
+   {{t 'Online and in-person event at '}}{{@data.event.locationName}}
+  </span>
+ {{else}}
+  <span class="{{if @orderPage 'weight-700'}}">{{t 'Online event'}}</span>
+ {{/if}}
+{{else if @data.event.locationName}}
+ {{#if @orderPage}}
```

Comment:
This component should not have any knowledge of order page. 

Use `outlet` to add custom child components

Reply:
This is confusing.
Changes are needed in the child component, if I render it using `outlet` then after that how will changes be made to it in the parent component?
Also, `event-info.hbs` and `general-info.hbs` are in different folder locations.

Reply:
```hbs
<EventLocation @event={{event}}>
  <a href="{{href-to 'events.view.edit.basic-details'}}">{{t 'Click to add location.'}}</a>
</EventLocation>
```

`outlet` is used to render custom elements as children in a particular component 

As EventLocation should have no idea about order page or event page, the links like `Click to add location` can never be placed in them. They should be placed in respective order page or event page wherever the EventLocation component is used according to their need

Think, what would you do if it needs to show 14 different things on every different page. Will you add 14 if else statements. A component should not have logic according to its parent. It should be unaware of its parent. Parent should inject in outlet whatever it needs to

Reply:
Yes I understand this part. But the problem is
```
{{#if @data.event.online}}
 {{#if @data.event.locationName}}
  <span class="{{if @orderPage 'weight-700'}}">
   {{t 'Online and in-person event at '}}{{@data.event.locationName}}
  </span>
 {{else}}
  <span class="{{if @orderPage 'weight-700'}}">{{t 'Online event'}}</span>
 {{/if}}
{{else if @data.event.locationName}}
 {{#if @orderPage}}
  <strong>{{t 'At '}} {{@data.event.locationName}}</strong>
 {{else}}
  <a href="{{href-to 'events.view.edit.basic-details'}}">{{t 'Click to add location.'}}</a>
 {{/if}}
{{else}}
  {{#if @orderPage}}
   <strong>{{t 'Location or online event details to be announced'}}</strong>
  {{else}}
   {{t 'To be announced.'}}
 {{/if}}
{{/if}}
```
Both these pages are displaying fairly the same information but are different in their code part, eg one is showing weighted text class other is not, secondly in one component `<strong>{{t 'Location or online event details to be announced'}}</strong>` is displayed while in other `{{t 'To be announced.'}}` is displayed. If I use outlet then probably I'll be writing the whole code again and injecting it as outlet and if I do this then what is the point of creating a component.

Reply:
I meant yield, not outlet https://guides.emberjs.com/release/tutorial/part-1/component-basics/